lib/riemann/tools/riemann_client_wrapper.rb

```diff
@@ -47,12 +47,18 @@ def configure(options)
         end
         @worker.abort_on_exception = true
 
+        at_exit { drain }
+
         self
       end
 
       def <<(event)
         @queue << event
       end
```

Comment:
We should probably raise an exception when we attempt to add events to the queue of a client in drain mode.

Reply:
Done!